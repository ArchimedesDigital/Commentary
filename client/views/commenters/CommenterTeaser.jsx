import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

// commenter Teaser
CommenterTeaser = React.createClass({

  propTypes: {
    commenter: React.PropTypes.object.isRequired
  },

  getChildContext() {
    return { muiTheme: getMuiTheme(baseTheme) };
  },

  childContextTypes: {
    muiTheme: React.PropTypes.object.isRequired,
  },

  render() {
    let commenter = this.props.commenter;
    let commenter_url = "/commenters/" + commenter.slug ;

     return (
       <div className="author-teaser hvr-grow wow fadeIn" >
              <a href="/commenter/show/{commenter.id}" >
                  <div className="author-image paper-shadow">
                    <img src="/images/default_user.jpg" alt="{commenter.name}"/>
                  </div>
              </a>
              <div className="author-teaser-text">
                  <a href="/commenters/show/{commenter.id}" >
                      <h3>{commenter.name}</h3>
                  </a>
                  <hr/>
                  <p className="author-description">
                      {commenter.tagline}
                  </p>

              </div>
          </div>


      );
    }

});
