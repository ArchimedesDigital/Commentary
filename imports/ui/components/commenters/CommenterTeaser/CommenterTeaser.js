import React from 'react';

import getMuiTheme from 'material-ui/styles/getMuiTheme';
import AvatarIcon from '/imports/ui/components/avatar/AvatarIcon';
import Utils from '/imports/lib/utils';

// lib:
import muiTheme from '/imports/lib/muiTheme';

// commenter Teaser
const CommenterTeaser = React.createClass({

	propTypes: {
		commenter: React.PropTypes.object.isRequired,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(muiTheme) };
	},

	render() {
		const commenter = this.props.commenter;
		const commenterUrl = `/commenters/${commenter.slug}`;
		const commenterExcerpt = commenter.tagline ? Utils.trunc(commenter.tagline, 120) : '';


		return (
			<div className="commenter-teaser hvr-grow wow fadeIn">
				<a href={commenterUrl}>
					<div className="commenter-image paper-shadow">
						{commenter && commenter.avatar ?
							<AvatarIcon avatar={commenter.avatar.src} />
						:
							<img
								src={commenter && commenter.avatar ? commenter.avatar : '/images/default_user.jpg'}
								alt={commenter.name}
							/>
						}
					</div>
				</a>
				<div className="commenter-teaser-text">
					<a href={commenterUrl}>
						<h3>{commenter.name}</h3>
					</a>
					<hr />
					<p className="commenter-description">
						{commenterExcerpt}
					</p>

				</div>
			</div>


		);
	},

});


export default CommenterTeaser;
