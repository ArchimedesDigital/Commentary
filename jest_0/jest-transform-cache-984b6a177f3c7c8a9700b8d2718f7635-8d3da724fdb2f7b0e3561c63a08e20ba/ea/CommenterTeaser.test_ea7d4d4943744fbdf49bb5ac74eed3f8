/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_CommenterTeaser = require('./CommenterTeaser'); /* istanbul ignore next */var _CommenterTeaser2 = _interopRequireDefault(_CommenterTeaser);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('CommenterTeaser', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_CommenterTeaser2.default, /* istanbul ignore next */{
			commenter: {
				name: 'Test Commenter',
				slug: 'test-commenter' } })).



		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: