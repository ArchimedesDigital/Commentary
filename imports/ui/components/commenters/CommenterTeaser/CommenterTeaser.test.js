import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommenterTeaser from './CommenterTeaser';

describe('CommenterTeaser', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<CommenterTeaser
					commenter={{
						name: 'Test Commenter',
						slug: 'test-commenter',
					}}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
