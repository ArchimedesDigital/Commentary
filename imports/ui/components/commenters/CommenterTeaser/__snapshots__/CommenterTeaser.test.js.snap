// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommenterTeaser renders correctly 1`] = `
<div
  className="commenter-teaser hvr-grow wow fadeIn"
>
  <a
    href="/commenters/test-commenter"
  >
    <div
      className="commenter-image paper-shadow"
    >
      <img
        alt="Test Commenter"
        src="/images/default_user.jpg"
      />
    </div>
  </a>
  <div
    className="commenter-teaser-text"
  >
    <a
      href="/commenters/test-commenter"
    >
      <h3>
        Test Commenter
      </h3>
    </a>
    <hr />
    <p
      className="commenter-description"
    >
      
    </p>
  </div>
</div>
`;
